File: Unstop/Practice Problems/Easy/Mary and Flowers/maryandflowers.py
def find_flowers(N, t, a):
    i, j = 0, N-1

    while i < j:
        total = a[i]+a[j]
        if total == t:
            return i, j
        elif total < t:
            i += 1
        else:
            j -= 1
    return None


N, t = map(int, input().split())
a = list(map(int, input().split()))

result = find_flowers(N, t, a)
if result:
    print(*result)
else:
    print("No valid pair found.")

File: Unstop/Practice Problems/Easy/Mary and Flowers/maryandflowers.cpp
#include <iostream>
#include <vector>
using namespace std;

pair<int, int> find_flower_indices(int N, int t, vector<int> &a)
{
    // Initialize two pointers
    int i = 0, j = N - 1;

    // Search for pair sum equal to 't'
    while (i < j)
    {
        int total = a[i] + a[j];
        if (total == t)
        {
            return {i, j};
        }
        else if (total < t)
        {
            i++;
        }
        else
        {
            j--;
        }
    }

    // No valid pair found
    return {-1, -1};
}

int main()
{
    int N, t;
    cin >> N >> t;
    vector<int> a(N);
    for (int i = 0; i < N; i++)
    {
        cin >> a[i];
    }

    auto result = find_flower_indices(N, t, a);
    if (result.first != -1)
    {
        cout << result.first << " " << result.second;
    }
    else
    {
        cout << "No valid pair found.";
    }

    return 0;
}
